cloud/services/net/net.go

```diff
@@ -45,22 +48,92 @@ func (s *Service) CreateNet(spec *infrastructurev1beta1.OscNet, clusterName stri
 	}
 	oscApiClient := s.scope.GetApi()
 	oscAuthClient := s.scope.GetAuth()
-	netResponse, httpRes, err := oscApiClient.NetApi.CreateNet(oscAuthClient).CreateNetRequest(netRequest).Execute()
-	if err != nil {
-		fmt.Printf("Error with http result %s", httpRes.Status)
-		return nil, err
+	var netResponse osc.CreateNetResponse
+	createNetCallBack := func() (bool, error) {
+		var httpRes *_nethttp.Response
+		var err error
+		netResponse, httpRes, err = oscApiClient.NetApi.CreateNet(oscAuthClient).CreateNetRequest(netRequest).Execute()
+		if err != nil {
+			if httpRes != nil {
+				fmt.Printf("Error with http result %s", httpRes.Status)
+			}
+			requestStr := fmt.Sprintf("%v", netRequest)
+			if reconciler.KeepRetryWithError(
+				requestStr,
+				httpRes.StatusCode,
+				reconciler.AllowedErrors) {
+				return false, nil
+			}
+			return false, fmt.Errorf("could not create net in Osc: %v", err)
+		}
+		return true, err
+	}
+	backoff := reconciler.EnvBackoff()
+	waitErr := wait.ExponentialBackoff(backoff, createNetCallBack)
+	if waitErr != nil {
+		return nil, waitErr
 	}
 	resourceIds := []string{*netResponse.Net.NetId}
-	err = tag.AddTag("Name", netName, resourceIds, oscApiClient, oscAuthClient)
-	if err != nil {
-		fmt.Printf("Error with http result %s", httpRes.Status)
-		return nil, err
+	netTag := osc.ResourceTag{
+		Key:   "Name",
+		Value: netName,
 	}
-	err = tag.AddTag("OscK8sClusterID/"+clusterName, "owned", resourceIds, oscApiClient, oscAuthClient)
-	if err != nil {
-		fmt.Printf("Error with http result %s", httpRes.Status)
+	netTagRequest := osc.CreateTagsRequest{
+		ResourceIds: resourceIds,
+		Tags:        []osc.ResourceTag{netTag},
+	}
+
+	addTagNameCallBack := func() (bool, error) {
+		err, httpRes := tag.AddTag(netTagRequest, resourceIds, oscApiClient, oscAuthClient)
+		if err != nil {
+			if httpRes != nil {
+				fmt.Printf("Error with http result %s", httpRes.Status)
+			}
+			requestStr := fmt.Sprintf("%v", netTagRequest)
+			if reconciler.KeepRetryWithError(
+				requestStr,
+				httpRes.StatusCode,
+				reconciler.AllowedErrors) {
+				return false, nil
+			}
+			return false, fmt.Errorf("%w failed to add Name tag", err)
+		}
+		return true, err
+	}
+	waitErr = wait.ExponentialBackoff(backoff, addTagNameCallBack)
+	if waitErr != nil {
 		return nil, err
 	}
```

Comment:
I did see during the first review but the retry should be done when you make the call API.

Here the retry should be done inside the `tag.AddTag`.

With you current changes, we have a lot of duplicate codes !

Reply:
Ok done